plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -180,13 +289,13 @@ instance Show (BuiltinRuntime (CekValue uni fun)) where
 -- 'Values' for the modified CEK machine.
 data CekValue uni fun =
     VCon (Some (ValueOf uni))
-  | VDelay (Term Name uni fun ()) (CekValEnv uni fun)
-  | VLamAbs Name (Term Name uni fun ()) (CekValEnv uni fun)
+  | VDelay (ETerm uni fun) (CekValEnv uni fun)
+  | VLamAbs !Unique (ETerm uni fun) (CekValEnv uni fun)
```

Comment:
Do you need a completely new type of terms to replace names with uniques?

Reply:
Well, this way GHC knows statically that the name is a `Unique` and can unpack it and also just pass it directly to `IntMap` functions. I think it's doing that automatically (or at least, adding UNPACK pragmas here made no difference).